librz/include/rz_util/rz_path.h

```diff
@@ -30,6 +31,15 @@ RZ_API RZ_OWN char *rz_path_home_expand(RZ_NULLABLE const char *path);
 
 RZ_API RZ_OWN char *rz_path_realpath(RZ_NULLABLE const char *path);
 
+typedef struct rz_path_portable {
+	char *prefix;
+	bool prefix_searched;
+	RzThreadLock *prefix_mutex;
+} RzPathPortable;
```

Comment:
Call this just RzPath and make it a private structure. (see how RzDiff is defined)